File: go/gigasecond/gigasecond.go
package gigasecond

import "time"

const TestVersion = 1

var Birthday = time.Date(1983, 11, 2, 9, 31, 0, 0, time.UTC)

func AddGigasecond(start time.Time) time.Time {
	return start.Add(1e9 * time.Second)
}

File: scala/gigasecond/src/main/scala/Gigasecond.scala
import java.util.Calendar.SECOND
import java.util.GregorianCalendar

/**
  *
  * @author Cedric Zhuang
  *
  */
case class Gigasecond(gc: GregorianCalendar) {
  lazy val date: GregorianCalendar = {
    val ret = gc.clone.asInstanceOf[GregorianCalendar]
    ret.add(SECOND, 1e9.toInt)
    ret
  }
}
